### PR TITLE
Add snapped and clicked outlet point

### PR DESCRIPTION
**Pull request by mmcfarland**
Preserve the originally clicked location and the point which the RWD
service snapped to along with the AoI.

* Upgrade RWD service to that which supports returning outlet points
* Save click and snap points and display on map alongside AoI

Connects #1085 

##### Caveats
* Marker style, marker duration and popup language open to feedback, intending to release on staging for experiments

##### Test
* Provision worker
* Delineate a watershed using Stream Network
* See that there are two points, one at your original click location.
* See that the markers are visible while the AoI is visible, and are removed when the AoI is reset or redrawn.

### CODE REVIEW

**Comment by mmcfarland** [comment]
@azavea-bot rebuild

**Comment by rajadain** [comment]
+1 tested

image

Saw the two map points in various stream networks. Clicking on each describes it in a popup, which is nice. Resetting draw tools removes the points as well as the area of interest. Delineating another one adds new points. Delineating using Hill Slope adds only the snapped point, but does have the popup which is useful:

image

**Comment by mmcfarland** [comment]
@azavea-bot rebuild

File: src/mmw/js/src/draw/views.js [hunk 1]
```diff
@@ -410,10 +412,32 @@ var WatershedDelineationView= Marionette.ItemView.extend({
                 self.rwdTaskModel.start(taskHelper);
                 return deferred;
             })
-            .done(validateShape)
-            .done(function(shape) {
-                addLayer(shape, itemName);
-                navigateToAnalyze();
+            .done(_.partialRight(validateShape, 'watershed'))
```
Comment (rajadain) [inline src/mmw/js/src/draw/views.js]: Cool use of `_.partialRight`

**Comment by rajadain** [comment]
Code looks great! Very clean.

**Pull request merged**